lighthouse-core/runner.js

```diff
@@ -289,11 +289,13 @@ class Runner {
       for (const artifactName of audit.meta.requiredArtifacts) {
         const noArtifact = artifacts[artifactName] === undefined;
 
-        // If trace required, check that DEFAULT_PASS trace exists.
-        // TODO: need pass-specific check of networkRecords and traces.
-        const noTrace = artifactName === 'traces' && !artifacts.traces[Audit.DEFAULT_PASS];
+        // If trace/devtoolsLog required, check that DEFAULT_PASS trace/devtoolsLog exists.
+        // TODO: need pass-specific check of traces and devtoolsLogs.
```

Comment:
> I would personally just get rid of this TODO, are we really going to move to `requiredArtifacts: ['traces.mySpecialPass']` at any point in the near future?

I think we're going to run into this sooner or later for the custom config/plugin reason you mention above. I think there are solutions, e.g. we could use a `tracePassName` (or whatever) convention in the audit options that defaults to `defaultPass` and check based on that here. That has its own pretty serious issues (we need to check `requiredArtifacts` *and* audit options now? what if an audit doesn't want to/can't follow the convention? etc), but I'd like to keep thinking about it.

Main thing is that if/when that situation does come up, this is where the error will occur, so it would be good to have a comment on it. I can definitely make it not a TODO, though.